cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -60,14 +60,13 @@ func (s *server) Start(ctx context.Context, host component.Host) error {
 	if err != nil {
 		return fmt.Errorf("cannot find primary storage %s: %w", s.config.Storage.TracesPrimary, err)
 	}
+	f = storage.NewDecoratorFactory(f, queryMetricsFactory)
 
 	spanReader, err := f.CreateSpanReader()
 	if err != nil {
 		return fmt.Errorf("cannot create span reader: %w", err)
 	}
 
-	spanReader = spanstoremetrics.NewReaderDecorator(spanReader, queryMetricsFactory)
```

Comment:
this is not the only place it's used, we need to remove it everywhere. And it would be better to change these methods to private, so that only the factory decorator can call them.

Reply:
@yurishkuro I left a question for you at https://github.com/jaegertracing/jaeger/pull/6267#discussion_r1862404219 regarding this.